### ISSUE TITLE
Getting Drew to commit to (a) Git repository/-ies

### ISSUE THREAD

**Issue by tarsius** [melpa/melpa#5034]
Many attempts have been made to convince Drew Adams to use `git` or `hg` to maintain his packages, but his setup works for him and he doesn't want to change it.

We have to respect that, but since we have decided to stop updateing packages from the Emacswiki and to eventually remove them, we have to find a solution to this. Drew is almost the only one who still uses the wiki to publish packages - almost all the other packages have not seen any updates in years.

Like we cannot force Drew to us version control, we cannot be expected to maintain some special support just for his packages. (Anymore. I have spend many many hours in the past, maintaining special cases for the sake of his packages (mostly as part of my work on the Emacsmirror). I am done with that.)

But he is the author of many popular packages, so we also cannot just remove those. So we have to find a compromise.

I suspect Drew uses something like `yaoddmuse-post-library` to upload to the Emacswiki. That could be replaced with:

```lisp
(defun drew-publish-this-file ()
  (yaoddmuse-post-library ...)
  (drew-commit-and-push ...))
```

Anyone willing to convince Drew to use something like this and to work with him to implement it?

**Comment by purcell** [comment]
I personally am fine with removing those packages unconditionally, so we should time-box this IMO, otherwise it will hold everything up.

**Comment by tarsius** [comment]
Oh, I agree. But maybe someone volunteers and then also succeeds in convincing Drew. Cannot hurt to ask.

**Comment by Silex** [comment]
An alternative could be some travis-ci job that does emacswiki-to-git conversion? Like everyday, fetch the emacswiki file over the one inside some git repo, commit & push?

Of course I guess the problem is not writing such hacks, but to maintain it.

**Comment by purcell** [comment]
> An alternative could be some travis-ci job that does emacswiki-to-git conversion? Like everyday, fetch the emacswiki file over the one inside some git repo, commit & push?

Well, we already have (had?) code which fetches code from the emacswiki, and that's exactly what we're trying to get rid of. We therefore won't be willing to host anything that was taken unconditionally from Emacswiki, even if it's placed in a source which superficially appears more trustworthy. :-)

**Comment by Silex** [comment]
Yeah I understand. This recurrent problem in various communities of "this workflow works for me" is hard to solve.

It makes sense for these established contributors to get things their way because they are the main contributors, but at the same time it prevents new contributors and gets the projects "stuck" from entering modern eras.

I've seen this for Emacs, Boost and Git. Always on different subjects, but always the same patterns of refusing to evolve to better paradigms on one side and refusing to adapt to old paradigms on the other side.

**Comment by alphapapa** [comment]
IMHO, if the users of those packages want them badly enough, they will install them manually from the wiki.  Otherwise, perhaps removing them will encourage someone to step up and maintain them in a repo.

**Comment by kaushalmodi** [comment]
As much as I respect Drew and his packages, and also use quite a many of those, I believe we should just do the right thing and move forward. @tarsius I'll try emailing Drew and see if he's fine with what you suggest.

**Comment by tarsius** [comment]
Could you please summarize here "what I (tarsius) have suggested" before you do so?

What I would like to ask him is "If someone patched the tool that you use to upload to the Emacswiki to upload to the Emacswiki *and* commit and push using Git, would you we willing to use that and help in the process of making that work for you".

**Comment by tarsius** [comment]
Here is a very early draft: https://gist.github.com/tarsius/31888dfe41a938d619b0c4ef195b00dd.

**Comment by Silex** [comment]
@tarsius: did you create this in *4 minutes* :astonished: ?

**Comment by tarsius** [comment]
No!

**Comment by Silex** [comment]
:sweat_smile: ah, I was worried I had to start a cult devoted to tarsius, the productivity deity.

Anyway, the draft looks good!

**Comment by alphapapa** [comment]
That's too bad.  Thanks for trying, Jonas.

I agree with @kaushalmodi that we should go ahead and remove the packages.  Those who want or need them can still install them with other tools.
